middlewares/taskRequests.js

```diff
@@ -0,0 +1,46 @@
+const { SOMETHING_WENT_WRONG } = require("../constants/errorMessages");
+const { userState } = require("../constants/userStatus");
+const userModel = require("../models/users.js");
+const userStatusModel = require("../models/userStatus.js");
+
+/**
+ * Validates user id for task request
+ *
+ * @param userId { string }: user id of the user
+ */
+async function validateUser(req, res, next) {
+  try {
+    const { userId } = req.body;
+
+    if (!userId) {
+      return res.boom.badRequest("userId not provided");
+    }
+
+    const { userExists, user } = await userModel.fetchUser({ userId });
+    if (!userExists) {
+      return res.boom.conflict("User does not exist");
+    }
+
+    const { userStatusExists, data: userStatus } = await userStatusModel.getUserStatus(userId);
+    if (!userStatusExists) {
+      return res.boom.conflict("User status does not exist");
+    }
+    if (userStatus.currentStatus.state === userState.OOO) {
```

Comment:
will return a 409 Conflict error if the user's current status is either OOO or ACTIVE. However, it is possible for a user to be in a state where they are neither OOO nor ACTIVE. For example, a user could be in a state where they are available but have not yet started working on a task.

Reply:
We only want to allow users with `IDLE` status to apply for a task. There may be a case where a user has started working on a task but that task has some blockers, then the task status would be marked as `Blocked` and the user status would be `IDLE` hence allowing the user to request for a new task.